File: ros2_ws/src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

// Add path `/opt/ros/humble/include/**` to vs code path

class MyNode: public rclcpp::Node {
    public:
    MyNode(const char* name): Node(name) {
        RCLCPP_INFO(this->get_logger(), name);
        timer = this->create_wall_timer(std::chrono::seconds(1),
                                        std::bind(&MyNode::timerCallback, this));
    }

    private:
    int counter{};
    rclcpp::TimerBase::SharedPtr timer;
    void timerCallback(void) {
        RCLCPP_INFO(this->get_logger(), "timerCallback %d", counter++);
    }
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>("cpp_test");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self, name):
        super().__init__(name)
        self.counter = 0
        self.get_logger().info("Node __init__")
        self.create_timer(0.5, self.timer_callback)
    def timer_callback(self):
        self.counter += 1
        self.get_logger().info(f'timer_callback {self.counter}')

def main(args=None):
    rclpy.init(args=args)
    node = MyNode('py_test')

    node.get_logger().info("ROS2 Node")

    rclpy.spin(node)
    rclpy.shutdown() # Last line

if __name__ == '__main__':
    main()